src/libraries/System.Private.CoreLib/src/System/SpanHelpers.cs

```diff
@@ -12,322 +17,184 @@ namespace System
-                    Unsafe.As<byte, long>(ref Unsafe.Add<byte>(ref b, 8)) = 0;
-#else
-                    Unsafe.As<byte, int>(ref b) = 0;
-                    Unsafe.As<byte, int>(ref Unsafe.Add(ref b, 4)) = 0;
-                    Unsafe.As<byte, int>(ref Unsafe.Add(ref b, 8)) = 0;
-                    Unsafe.As<byte, int>(ref Unsafe.Add(ref b, 12)) = 0;
-#endif
-                    return;
-                case 17:
-#if TARGET_64BIT
-                    Unsafe.As<byte, long>(ref b) = 0;
-                    Unsafe.As<byte, long>(ref Unsafe.Add<byte>(ref b, 8)) = 0;
+            ref byte bEnd = ref Unsafe.Add(ref b, byteLength);
+
+            if (byteLength <= 16)
+                goto MZER02;
+            if (byteLength > 64)
+                goto MZER05;
+
+        MZER00:
+            // Clear bytes which are multiples of 16 and leave the remainder for MZER01 to handle.
+            Debug.Assert(byteLength > 16 && byteLength <= 64);
+#if HAS_CUSTOM_BLOCKS
+            Unsafe.WriteUnaligned<Buffer.Block16>(ref b, default); // [0,16]
+#elif TARGET_64BIT
+        Unsafe.WriteUnaligned<long>(ref b, 0);
+            Unsafe.WriteUnaligned<long>(ref Unsafe.Add(ref b, 8), 0); // [0,16]
 #else
-                    Unsafe.As<byte, int>(ref b) = 0;
-                    Unsafe.As<byte, int>(ref Unsafe.Add(ref b, 4)) = 0;
-                    Unsafe.As<byte, int>(ref Unsafe.Add(ref b, 8)) = 0;
-                    Unsafe.As<byte, int>(ref Unsafe.Add(ref b, 12)) = 0;
+            Unsafe.WriteUnaligned<int>(ref b, 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 4), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 8), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 12), 0); // [0,16]
 #endif
-                    Unsafe.Add(ref b, 16) = 0;
-                    return;
-                case 18:
-#if TARGET_64BIT
-                    Unsafe.As<byte, long>(ref b) = 0;
-                    Unsafe.As<byte, long>(ref Unsafe.Add<byte>(ref b, 8)) = 0;
+            if (byteLength <= 32)
+                goto MZER01;
+#if HAS_CUSTOM_BLOCKS
+        Unsafe.WriteUnaligned<Buffer.Block16>(ref Unsafe.Add(ref b, 16), default); // [0,32]
+#elif TARGET_64BIT
+            Unsafe.WriteUnaligned<long>(ref Unsafe.Add(ref b, 16), 0);
+        Unsafe.WriteUnaligned<long>(ref Unsafe.Add(ref b, 24), 0); // [0,32]
 #else
-                    Unsafe.As<byte, int>(ref b) = 0;
-                    Unsafe.As<byte, int>(ref Unsafe.Add(ref b, 4)) = 0;
-                    Unsafe.As<byte, int>(ref Unsafe.Add(ref b, 8)) = 0;
-                    Unsafe.As<byte, int>(ref Unsafe.Add(ref b, 12)) = 0;
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 16), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 20), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 24), 0);
+            Unsafe.WriteUnaligned<int>(ref Unsafe.Add(ref b, 28), 0); // [0,32]
```

Comment:
Do these comments add value?

This block is actually [16,32]

Reply:
Thanks, I don't think they do, removed